io/net/bonding.py

```diff
@@ -116,16 +116,46 @@ def setUp(self):
         if self.host_interface[0:2] == 'ib':
             self.ib = True
         self.log.info("Bond Test on IB Interface? = %s", self.ib)
-        self.session = Session(self.peer_first_ipinterface, user=self.user,
-                               password=self.password)
+
+        '''
+        An individual interface, that has a LACP PF, cannot communicate without
+        being bonded. So the test uses the public ip address to create an SSH
+        session instead of the private one when setting up a bonding interface.
+        '''
+        if self.mode == "4" and "setup" in str(self.name.name):
+            self.session = Session(self.peer_public_ip, user=self.user,
+                                   password=self.password)
+        else:
+            self.session = Session(self.peer_first_ipinterface, user=self.user,
+                                   password=self.password)
+
         if not self.session.connect():
-            self.cancel("failed connecting to peer")
+            '''
+            LACP bond interface takes some time to get it to ping peer after it
+            is setup. This code block tries at most 5 times to get it to connect
+            to the peer.
+            '''
+            if self.mode == "4":
+                connect = False
+                for _ in range(5):
+                    if self.session.connect():
+                        connect = True
+                        self.log.info("Was able to connect to peer.")
+                        break
+                    time.sleep(5)
+                if not connect:
+                    self.cancel("failed connecting to peer")
+            else:
+                self.cancel("failed connecting to peer")
```

Comment:
@abdhaleegit added this code block to fix the failing of connecting to peer. Let me know what you think!

Reply:
looks good